server.go

```diff
@@ -167,16 +167,21 @@ func (server *mongoServer) Connect(timeout time.Duration) (*mongoSocket, error)
 	case !dial.isSet():
 		// Cannot do this because it lacks timeout support. :-(
 		//conn, err = net.DialTCP("tcp", nil, server.tcpaddr)
-		conn, err = net.DialTimeout("tcp", server.ResolvedAddr, timeout)
-		if tcpconn, ok := conn.(*net.TCPConn); ok {
-			tcpconn.SetKeepAlive(true)
-		} else if err == nil {
-			panic("internal error: obtained TCP connection is not a *net.TCPConn!?")
+		conn, err = net.DialTimeout(server.raddr.Network(), server.ResolvedAddr, timeout)
+
+		switch conn.(type) {
```

Comment:
```
switch connImpl := conn.(type) {
case *net.UnixConn:
case *net.TCPConn:
connImpl.SetKeepAlive(true)
[...]
}
```

Reply:
I don't see the benefit here. Besides it adds a pointer for the GC to collect.